dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java

```diff
@@ -461,10 +462,23 @@ tea, getNumericClause(), getDateClause(), true))
         columns.add(EventAnalyticsColumn.TRACKED_ENTITY_GEOMETRY);
       }
     }
+    if (sqlBuilder.supportsDeclarativePartitioning()) {
+      // Add the year column required for declarative partitioning
+      columns.add(getPartitionColumn());
+    }
 
     return filterDimensionColumns(columns);
   }
 
+  @Override
+  protected AnalyticsTableColumn getPartitionColumn() {
+    return AnalyticsTableColumn.builder()
+        .name("year")
+        .dataType(INTEGER)
+        .selectExpression("CAST(SUBSTRING(dps.yearly, 1, 4) AS INTEGER)")
```

Comment:
Can we use resource table instead? Just feels more consistent with existing code and natural.